src/details/ArborX_Polytope.hpp

```diff
@@ -0,0 +1,270 @@
+    {
+      _min_values[i] = KokkosExt::ArithmeticTraits::max<float>::value;
+      _max_values[i] = -KokkosExt::ArithmeticTraits::max<float>::value;
+    }
+  }
+  KOKKOS_FUNCTION Polytope &operator+=(Point const &p)
+  {
+    using KokkosExt::max;
+    using KokkosExt::min;
+    for (int i = 0; i < n_directions; ++i)
+    {
+      auto const proj_i = Details::project(p, this->directions[i]);
+      _min_values[i] = min(_min_values[i], proj_i);
+      _max_values[i] = max(_max_values[i], proj_i);
+    }
+    return *this;
+  }
+  KOKKOS_FUNCTION Polytope &operator+=(Polytope const &other)
+  {
+    using KokkosExt::max;
+    using KokkosExt::min;
+    for (int i = 0; i < n_directions; ++i)
+    {
+      _min_values[i] = min(_min_values[i], other._min_values[i]);
+      _max_values[i] = max(_max_values[i], other._max_values[i]);
+    }
+    return *this;
+  }
+  KOKKOS_FUNCTION bool intersects(Point const &p) const
+  {
+    for (int i = 0; i < n_directions; ++i)
+    {
+      auto const proj_i = Details::project(p, this->directions[i]);
+      if (proj_i < _min_values[i] || proj_i > _max_values[i])
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+  KOKKOS_FUNCTION bool intersects(Box const &box) const
+  {
+    // using KokkosExt::minmax;
+    // for (int i = 0; i < n_directions; ++i)
+    // {
+    //   auto const minmax_proj_i =
+    //       minmax(Details::project(box.minCorner(), this->directions[i]),
+    //              Details::project(box.maxCorner(), this->directions[i]));
+    //   auto const min_proj_i = minmax_proj_i.first;
+    //   auto const max_proj_i = minmax_proj_i.second;
+    //   if (max_proj_i < _min_values[i] || min_proj_i > _max_values[i])
+    //   {
+    //     return false;
+    //   }
+    // }
+    // return true;
+    Polytope other{};
+    auto x0 = box.minCorner()[0];
+    auto x1 = box.minCorner()[1];
+    auto x2 = box.minCorner()[2];
+    auto y0 = box.maxCorner()[0];
+    auto y1 = box.maxCorner()[1];
+    auto y2 = box.maxCorner()[2];
+    for (auto const &point : {
+             Point{x0, x1, x2},
+             Point{x0, y1, x2},
+             Point{x0, x1, y2},
+             Point{x0, y1, y2},
+             Point{y0, x1, x2},
+             Point{y0, y1, x2},
+             Point{y0, x1, y2},
+             Point{y0, y1, y2},
+         })
+    {
+      other += point;
+    }
+    return intersects(other);
+  }
+  KOKKOS_FUNCTION bool intersects(Polytope const &other) const
+  {
+    for (int i = 0; i < n_directions; ++i)
+    {
+      if (other._max_values[i] < _min_values[i] ||
+          other._min_values[i] > _max_values[i])
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+  KOKKOS_FUNCTION Box toBox() const
+  {
+    // WARNING
+    Box b{};
+    for (int i = 0; i < 3; ++i)
+    {
+      b.minCorner()[i] = _min_values[i];
+      b.maxCorner()[i] = _max_values[i];
+    }
+    return b;
+  }
+};
+
+template <int k>
+struct DOP;
+
+template <>
+struct DOP<3>
```

Comment:
Each k in the kDOP paper actually corresponds to a direction, but not bidirectional. On page 4 they state:
```
To keep the associated costs as small as possible, we
have been using only k-dops whose discrete orientation
normals come as pairs of collinear, but oppositely oriented,
vectors.```
So, AABB is actually 6-dop.

I suggest following that terminology for consistency.